pkg/build/interfaces.go

```diff
@@ -49,6 +49,12 @@ type Ignorer interface {
 	Ignore(*api.Config) error
 }
 
+// Runner allows the user to actually run the produced imgae as part of the s2i
+// cmd line invocation, to see if it operates as he would expect
+type Runner interface {
+	Run(*api.Config) error
+}
```

Comment:
OK, I see the sub directory point.  It could be moved to an interfaces.go file in another directory.

Use of interfaces can be warranted not just for the current situation, but for the anticipation of needing future implementations in the future.  And if you defer on creating the interface until that need arises, your *initial* implementation often needs some rework to keep it in line with the implicit semantics, whereas explicitly defining them helps scope the situation.